src/test/java/seedu/placebook/logic/parser/AddAppCommandParserTest.java

```diff
@@ -0,0 +1,100 @@
+package seedu.placebook.logic.parser;
+
+import static seedu.placebook.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.placebook.logic.parser.CommandParserTestUtil.assertParseSuccess;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.placebook.commons.core.index.Index;
+import seedu.placebook.logic.commands.AddAppCommand;
+import seedu.placebook.model.person.Address;
+
+public class AddAppCommandParserTest {
+    private ArrayList<Index> indexes = new ArrayList<>();
+    private AddAppCommandParser parser = new AddAppCommandParser();
+
+    @Test
+    public void parse_allFieldsPresent_success() {
+        indexes.add(Index.fromZeroBased(0));
+        // whitespace only preamble
+        assertParseSuccess(parser,
+                " id/1 a/vivocity start/01-01-2021 1000 end/01-01-2021 1200 ds/Halloween Sales",
+                new AddAppCommand(indexes, new Address("vivocity"),
+                        LocalDateTime.of(2021, 1, 1, 10, 0),
+                        LocalDateTime.of(2021, 1, 1, 12, 0),
+                        "Halloween Sales"));
+    }
+
+    @Test
+    public void parse_optionalFieldsMissing_success() {
```

Comment:
Technically description isnt an optionalField

Reply:
Updated the Tests based on feedback!